dictionary/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/labstack/echo"
+	"github.com/labstack/echo/engine"
+	"github.com/labstack/echo/engine/standard"
+)
+
+type dictionaryEntry struct {
+	Word         string
+	Translations []string
+	Idioms       []string
+}
+
+var db *sql.DB
+
+// Creates new instance of dictionaryEntry and initializes its fields
+func newDictEntry() *dictionaryEntry {
+	var de = new(dictionaryEntry)
+	de.Idioms = make([]string, 0, 10)
+	de.Translations = make([]string, 0, 10)
+	return de
+}
+
+func addWord(c echo.Context) error {
+	var dictEntry = newDictEntry()
+	c.Bind(dictEntry)
+	var user = c.Request().Header().Get("User")
+	err := addDictEntry(db, user, *dictEntry)
```

Comment:
here and next your code has a higher coupling and adds complexity in tests (due to using global var db)